origamiUROP/polygons.py

```diff
@@ -60,33 +65,104 @@ class BoundaryPolygon:
     def __init__(self, vertices: np.ndarray):
         """ 4 or more vertex objects given, each value on a given row """
 
+        assert vertices.shape[1] == 3 or vertices.shape[1] == 2
         self.vertices = vertices
-        self.x = vertices[:, 0]
-        self.y = vertices[:, 1]
-        self.z = vertices[:, 2]
-        self.n_edges = self.vertices.shape[0]  # no. of sides = no of rows
 
-        self.edges = []
+    @property
+    def n_edges(self) -> np.array:
+        return self.vertices.shape[0]
 
-        # makes a index which "wraps around on itself, e.g. 4 edges -> [0,1,2,3,0]"
-        counter = np.append(np.arange(0, self.n_edges), 0)
+    @property
+    def edges(self) -> List[Edge]:
+        """
+        Iterate over all vertices and return a list of Edge instances
+        """
+        edges = []
         for i in range(self.n_edges):
-            self.edges.append(
-                define_edges(self.vertices, counter[i], counter[i + 1], 0)
+            edges.append(
+                Edge(self.vertices[i-1], self.vertices[i], 0)
```

Comment:
AH I'm so silly, I spent ages getting my head around how to wrap the list around on itself and [-1] just gives the last value! Oh well, you live and you learn hahah